src/tuf/index.ts

```diff
@@ -19,77 +19,81 @@ import { Updater } from 'tuf-js';
 import * as sigstore from '../types/sigstore';
 import { TrustedRootFetcher } from './trustroot';
 
-interface RepositoryMap {
-  repositories: Record<string, string[]>;
-  mapping: {
-    paths: string[];
-    repositories: string[];
-    threshold: number;
-    terminating: boolean;
-  }[];
+const DEFAULT_MIRROR_URL = 'https://sigstore-tuf-root.storage.googleapis.com';
+const DEFAULT_ROOT_PATH = '../../store/public-good-instance-root.json';
+
+export interface TUFOptions {
+  mirrorURL?: string;
+  rootPath?: string;
+}
+
+interface RemoteConfig {
+  mirror: string;
 }
 
 export async function getTrustedRoot(
-  cacheDir: string
+  cachePath: string,
+  options: TUFOptions = {}
 ): Promise<sigstore.TrustedRoot> {
-  initTufCache(cacheDir);
-  const repoMap = initRepoMap(cacheDir);
+  initTufCache(cachePath, options.rootPath);
+  const mirror = initRemoteConfig(cachePath, options.mirrorURL);
+  const repoClient = initClient(cachePath, mirror);
 
-  const repoClients = Object.entries(repoMap.repositories).map(([name, urls]) =>
-    initClient(name, urls[0], cacheDir)
-  );
-
-  // TODO: Add support for multiple repositories. For now, we just use the first
-  // one (the production Sigstore TUF repository).
-  const fetcher = new TrustedRootFetcher(repoClients[0]);
+  const fetcher = new TrustedRootFetcher(repoClient);
   return fetcher.getTrustedRoot();
 }
 
-// Initializes the root TUF cache directory
-function initTufCache(cacheDir: string): void {
-  if (!fs.existsSync(cacheDir)) {
-    fs.mkdirSync(cacheDir, { recursive: true });
+// Initializes the TUF cache directory structure
+function initTufCache(cachePath: string, rootPath: string | undefined): string {
+  const targetsPath = path.join(cachePath, 'targets');
+  const cachedRootPath = path.join(cachePath, 'root.json');
+
+  if (!fs.existsSync(cachePath)) {
+    fs.mkdirSync(cachePath, { recursive: true });
   }
-}
 
-// Initializes the repo map (copying it to the cache root dir) and returns the
-// content of the repository map.
-function initRepoMap(rootDir: string): RepositoryMap {
-  const mapDest = path.join(rootDir, 'map.json');
+  if (!fs.existsSync(targetsPath)) {
+    fs.mkdirSync(targetsPath);
+  }
 
-  if (!fs.existsSync(mapDest)) {
-    const mapSrc = require.resolve('../../store/map.json');
-    fs.copyFileSync(mapSrc, mapDest);
+  if (!fs.existsSync(cachedRootPath)) {
+    const initialRootPath = rootPath || require.resolve(DEFAULT_ROOT_PATH);
+    fs.copyFileSync(initialRootPath, cachedRootPath);
```

Comment:
Will this work with `ncc`?

Reply:
Yeah, since the `require.resolve` points to a static path now, ncc detects the file access and re-writes this line:

```javascript
if (!fs_1.default.existsSync(cachedRootPath)) {
    const initialRootPath = rootPath || __nccwpck_require__.ab + "public-good-instance-root.json";
    fs_1.default.copyFileSync(initialRootPath, cachedRootPath);
}
```

The use of `__nccwpck_require__.ab` means that ncc found that file and copied it into the distribution bundle.